apps/api/src/service/fileService.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2024 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import formidable from 'formidable';
+import fs from 'fs';
+
+import { type PostgresDb } from '@/db/index.js';
+import { files } from '@/db/schemas/files.js';
+import logger from '@/logger.ts';
+import { failure, success, type AsyncResult } from '@/utils/results.js';
+import { eq } from 'drizzle-orm';
+import { PostgresTransaction, type FilesModel, type JoinedApplicationRecord } from './types.ts';
+
+/**
+ * Upload service provides methods for file DB access
+ * @param db - Drizzle Postgres DB Instance
+ */
+const filesSvc = (db: PostgresDb) => ({
+	createFile: async ({
+		file,
+		application,
+		type,
+		transaction,
+	}: {
+		file: formidable.File;
+		application: JoinedApplicationRecord;
+		type: 'SIGNED_APPLICATION' | 'ETHICS_LETTER';
```

Comment:
This `type` param is worth capturing in its own variable (`/service/types` or `data-model`? where ever you see fit)